crates/tabby-download/src/lib.rs

```diff
@@ -65,7 +65,10 @@ async fn download_file(url: &str, path: &Path) -> Result<()> {
 
     let mut bar = WrappedBar::new(0, url, false);
 
-    https::HTTPSHandler::get(url, &intermediate_filename, &mut bar, "").await?;
+    if let Err(e) = https::HTTPSHandler::get(url, &intermediate_filename, &mut bar, "").await {
+        eprintln!("Fetching model failed: {e}");
+        std::process::exit(1);
```

Comment:
there's no need to call `exit`, see handler code in line 82

Reply:
Not sure whether you want me to use `panic!` or `error!`

Reply:
propagating error object would be good